### PR TITLE
[CMSP-353] Add pantheon-wp-coding-standards

### PR DESCRIPTION
**Pull request by jazzsequence**
Adds our coding standards and fixes sniffs.

Also updates the circle lint workflow to use PHP 8

### CODE REVIEW

**Comment by jazzsequence** [comment]
> Interesting to note that behat fell down where phpunit did not. Is that a case we should cover with unit testing?

@pwtyler I think there are legit issues that were introduced by some of the fixes for PHPCS sniffs. I'm trying to put stuff back the way it was to see if it can get through Behat and then be more intentional about what I'm changing. But I do think that if things are failing, probably unit tests should cover those unless they are legitimately not able to be covered by functional tests.

**Comment by jazzsequence** [comment]
Nevermind, it's the upstream tests that are failing, not the Behat tests running from the plugin. 🤦

**Comment by pwtyler** [comment]
Will hold review until #406 is updated with same behat fixes and this PR is rebased.

**Comment by jazzsequence** [comment]
@pwtyler This PR removes the upstream behat tests (although they're still loaded as a repository, we just aren't running them). I think while https://github.com/pantheon-systems/pantheon-wordpress-upstream-tests/pull/62 is still WIP we can go ahead and merge this and move forward and re-add the upstream tests as those are fixed.

**Review by CdrMarks — APPROVED** [review]

File: object-cache.php [hunk 1]
```diff
@@ -1191,38 +1185,37 @@ public function build_client_parameters( $redis_server ) {
...
-				$redis_server = array(
-					'host'     => $_SERVER['CACHE_HOST'],
-					'port'     => $_SERVER['CACHE_PORT'],
-					'auth'     => $_SERVER['CACHE_PASSWORD'],
-					'database' => isset( $_SERVER['CACHE_DB'] ) ? $_SERVER['CACHE_DB'] : 0,
-				);
+				$redis_server = [
+					'host' => wp_strip_all_tags( $_SERVER['CACHE_HOST'] ),
+					'port' => isset( $_SERVER['CACHE_PORT'] ) ? wp_strip_all_tags( $_SERVER['CACHE_PORT'] ) : 0,
+					'auth' => isset( $_SERVER['CACHE_PASSWORD'] ) ? wp_strip_all_tags( $_SERVER['CACHE_PASSWORD'] ) : '',
```
Comment (timnolte) [inline object-cache.php:1191]: @jazzsequence FYI, just noting that this line is incorrect and the default value should have been `null` as I had indicated in my PR to fixed this. Also, the default port should not be `0` as that is completely invalid for Redis and should be `6379`.

https://github.com/pantheon-systems/wp-redis/pull/360

Comment (timnolte) [reply object-cache.php:1191]: @jazzsequence noting also that changes should have fixed the issues called out in https://github.com/pantheon-systems/wp-redis/issues/359 but in fact broke our patch which we will now have to redo.

File: wp-redis.php [hunk 2]
```diff
@@ -36,19 +38,19 @@ function wp_redis_get_info() {
...
-				'port'     => $_SERVER['CACHE_PORT'],
-				'auth'     => $_SERVER['CACHE_PASSWORD'],
-				'database' => isset( $_SERVER['CACHE_DB'] ) ? $_SERVER['CACHE_DB'] : 0,
-			);
+		if ( isset( $_SERVER['CACHE_HOST'] ) && isset( $_SERVER['CACHE_PORT'] ) && isset( $_SERVER['CACHE_PASSWORD'] ) && isset( $_SERVER['CACHE_DB'] ) ) {
+			$redis_server = [
+				'host' => sanitize_text_field( $_SERVER['CACHE_HOST'] ),
+				'port' => sanitize_text_field( $_SERVER['CACHE_PORT'] ),
+				'auth' => sanitize_text_field( $_SERVER['CACHE_PASSWORD'] ),
+				'database' => sanitize_text_field( $_SERVER['CACHE_DB'] ),
```
Comment (timnolte) [inline wp-redis.php:46]: @jazzsequence this was also incorrectly changed to no longer be providing a default value.

File: object-cache.php (hunk 1 above)
Comment (jazzsequence) [reply object-cache.php:1191]: These have been updated in #428. Thanks for digging this up.

**Pull request merged**